### PR TITLE
Adds validations to insert and update queries

### PR DESCRIPTION
**Pull request by joeconwaystk**
Saving onUpdate, onInsert and validate for ManagedObject for the next PR

### CODE REVIEW

File: lib/src/db/managed/validate.dart [hunk 1]
```diff
@@ -0,0 +1,531 @@
...
+    if (_options.every((v) => value != v)) {
+      errors.add("The value for '${property.name}' is invalid. Must be one of: ${_options.map((v) => "'$v'").join(",")}.");
+      return false;
+    }
+
+    return true;
+  }
+
+  static String _errorStringForOperation(ValidateOperation op) {
+    if (op == ValidateOperation.insert) {
```
Comment (anachlas) [inline lib/src/db/managed/validate.dart]: Should this use a switch?

Comment (joeconwaystk) [inline lib/src/db/managed/validate.dart]: yes, will change

**Comment by joeconwaystk** [comment]
Gonna work out of master now and tag

**Comment by anachlas** [comment]
Ok that sounds a lot cleaner.

**Pull request merged**